jme3-effects/src/main/resources/Common/MatDefs/Post/ColorContrast.frag

```diff
@@ -30,61 +30,56 @@
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-
 /**
  * Utilized by the ColorContrast.j3md to adjust the textures color contrast and brightness
  * based on a transfer function that uses a simple power law on color.rgb before processing those colors as a fragment
- * shader by the rasterizer.
+ * shader by the rasterizer, the color channels can be scaled independently again at the final pass using
+ * a scale factor from 0 to 1.0.
  *
- * Supports glsl100.
+ * Supports GLSL100 GLSL110 GLSL120 GLSL130.
  */
 
-
 //constant inputs from java source
-uniform float exp_r;
-uniform float exp_g;
-uniform float exp_b;
-
+uniform float m_exp_r;
+uniform float m_exp_g;
+uniform float m_exp_b;
 
-uniform float minBrightness;
-uniform float maxBrightness;
+//final scale values
+uniform float m_scale_r;
+uniform float m_scale_g;
+uniform float m_scale_b;
 
+uniform float m_minBrightness;
+uniform float m_maxBrightness;
 
 //container for the input from post.vert
 uniform sampler2D m_Texture;
 
-
 //varying input from post.vert vertex shader
 varying vec2 texCoord;
 
-
 void main() {
 
-
     //get the color from a 2d sampler.
     vec4 color = texture2D(m_Texture, texCoord);
 
-
     //apply the color transfer function.
 
-
     //1) apply brightness to color.rgb.
-    color.r = (color.r - minBrightness) / (maxBrightness - minBrightness);
-    color.g = (color.g - minBrightness) / (maxBrightness - minBrightness);
-    color.b = (color.b - minBrightness) / (maxBrightness - minBrightness);
-
-
-    color.r = max(color.r, 0.0);
-    color.g = max(color.g, 0.0);
-    color.b = max(color.b, 0.0);
-
+    color.rgb = (color.rgb - vec3(m_minBrightness)) / (vec3(m_maxBrightness) - vec3(m_minBrightness));
```

Comment:
The brightness value can exceed 1.0 by setting the minBrightness to 0 and maxBrightness to less than 1.